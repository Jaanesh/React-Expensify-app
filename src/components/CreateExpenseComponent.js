import React from 'react';
import {connect} from 'react-redux' ;
import {addExpense} from '../actions/expense.js';
import ExpenseForm from './ExpenseFormComponent';

/*
//props got he dispatch event due to 'connect'
const createExpenseComponentTemplate=(props)=>(
    <div>
        <h2>This is createExpenseTemplate</h2>
        <br/>
        <h2>Add New Expense</h2>
        <br/>
        <ExpenseForm
             onSubmit={(newExpense)=>{
                 console.log(newExpense);
                 props.dispatch(addExpense(newExpense));
                 props.history.push('/');
             }}></ExpenseForm>
    </div>

);

export default connect()(createExpenseComponentTemplate);
*/

export class CreateExpenseComponentTemplate extends React.Component{
   
    onSubmit=(newExpense)=>{
        console.log(newExpense);
        this.props.onSubmit(newExpense);
        this.props.history.push('/');
    }

    render(){
        return (
            <div>
            <h2>This is createExpenseTemplate</h2>
            <br/>
            <h2>Add New Expense</h2>
            <br/>
            <ExpenseForm onSubmit={this.onSubmit} ></ExpenseForm>
        </div>
        );
    }
}

const mapDispatchToProps=(dispatch)=>{
     return{
        onSubmit:(newExpense)=>dispatch(addExpense(newExpense))
     }
}

export default connect(undefined,mapDispatchToProps)(CreateExpenseComponentTemplate);