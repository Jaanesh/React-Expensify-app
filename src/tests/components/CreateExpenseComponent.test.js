import React from 'react';
import {shallow} from 'enzyme';
import moment from 'moment';
import {CreateExpenseComponentTemplate}  from '../../components/CreateExpenseComponent.js';

let expenseObj1={id:'1', description:'jan rent',note:'',amount:3,createdAt:moment(0).
                                                                    subtract(5,'days').valueOf()};
let wrapper,onSubmit,history;

beforeEach(()=>{
    onSubmit=jest.fn();
    history={push:jest.fn()};
    wrapper=shallow(<CreateExpenseComponentTemplate onSubmit={onSubmit} history={history} />);
});

test('should render CreateExpenseComponentTemplate correctly',()=>{
    expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit',()=>{
    wrapper.find('ExpenseForm').prop('onSubmit')(expenseObj1);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(onSubmit).toHaveBeenLastCalledWith(expenseObj1);
});
