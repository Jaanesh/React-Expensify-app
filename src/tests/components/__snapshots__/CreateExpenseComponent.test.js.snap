// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CreateExpenseComponentTemplate correctly 1`] = `
<div>
  <h2>
    This is createExpenseTemplate
  </h2>
  <br />
  <h2>
    Add New Expense
  </h2>
  <br />
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
